src/vehicle_gui.cpp

```diff
@@ -1611,27 +1810,47 @@ struct VehicleListWindow : public BaseVehicleListWindow {
 	{
 		switch (widget) {
 			case WID_VL_SORT_ORDER: // Flip sorting method ascending/descending
-				this->vehicles.ToggleSortOrder();
+				this->vehgroups.ToggleSortOrder();
 				this->SetDirty();
 				break;
 
+			case WID_VL_GROUP_BY_PULLDOWN:// Select sorting criteria dropdown menu
```

Comment:
Nitpick: missing space between `:` and `//` (fix the `case` below too while at it please :D)